File: app/src/main/java/com/weathercock/cleanarchitecture/di/module/DataModule.java
package com.weathercock.cleanarchitecture.di.module;

import android.content.Context;

import com.weathercock.cleanarchitecture.data.repository.AboutRepositoryImpl;
import com.weathercock.cleanarchitecture.data.repository.CityRepositoryImpl;
import com.weathercock.cleanarchitecture.data.repository.WeatherRepositoryImpl;
import com.weathercock.cleanarchitecture.domain.repository.AboutRepository;
import com.weathercock.cleanarchitecture.domain.repository.CityRepository;
import com.weathercock.cleanarchitecture.domain.repository.WeatherRepository;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class DataModule {

    @Provides
    @Singleton
    public CityRepository provideCityRepository() {
        return new CityRepositoryImpl();
    }

    @Provides
    @Singleton
    public WeatherRepository provideWeatherRepository() {
        return new WeatherRepositoryImpl();
    }

    @Provides
    @Singleton
    public AboutRepository provideAboutRepository(Context context) {
        return new AboutRepositoryImpl(context);
    }

}

File: data/src/main/java/com/kazakago/cleanarchitecture/data/di/DataModule.kt
package com.kazakago.cleanarchitecture.data.di

import com.kazakago.cleanarchitecture.data.repository.about.AboutRepositoryImpl
import com.kazakago.cleanarchitecture.data.repository.city.CityRepositoryImpl
import com.kazakago.cleanarchitecture.data.repository.weather.WeatherRepositoryImpl
import com.kazakago.cleanarchitecture.domain.repository.about.AboutRepository
import com.kazakago.cleanarchitecture.domain.repository.city.CityRepository
import com.kazakago.cleanarchitecture.domain.repository.weather.WeatherRepository
import org.koin.dsl.module.module

val dataModule = module {
    single<CityRepository> { CityRepositoryImpl(get()) }
    single<WeatherRepository> { WeatherRepositoryImpl(get()) }
    single<AboutRepository> { AboutRepositoryImpl(get()) }
}